src/main/java/io/stargate/sgv2/jsonapi/service/operation/collections/ReadAndUpdateCollectionOperation.java

```diff
@@ -163,61 +165,80 @@ private Uni<UpdatedDocument> processUpdate(
               boolean upsert = readDocument.txnId().isEmpty();
               JsonNode originalDocument = upsert ? null : readDocument.get();
 
-              // apply document updates
-              // if no changes return null item
               DocumentUpdater.DocumentUpdaterResponse documentUpdaterResponse =
                   documentUpdater().apply(readDocument.get().deepCopy(), upsert);
 
-              // In case no change to document and not an upsert document, short circuit and return
-              if (!documentUpdaterResponse.modified() && !upsert) {
-                // If no change return the original document Issue #390
-                if (returnDocumentInResponse) {
-                  resultProjection.applyProjection(originalDocument);
-                  return Uni.createFrom()
-                      .item(
-                          new UpdatedDocument(
-                              readDocument.id().orElseThrow(), upsert, originalDocument, null));
-                } else {
-                  return Uni.createFrom().nullItem();
-                }
-              }
+              return documentUpdaterResponse
```

Comment:
This is where it looked to be called multiple times so lazy cache of the vectorized data would help in case of UpdateMany.